### PR TITLE
Fixed a bug that could lead to perceived message loss under JetStream.

### PR DESCRIPTION
**Pull request by derekcollison**
Under load and pressure from concurrent publishing and consuming with multiple consumers the filestore would return a partial or no cache error to the upper layers. For consumers this could result in us skipping a stream sequence when we should not.

 This change stabilizes the filestore and removes the flush state for msg blocks. I also found some bugs that did not track last sequence properly after snapshots / restore.

Also increased timeout under Travis for the Go tests. We were bumping up against the 10m default.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

**Comment by anjmao** [comment]
Hi @derekcollison 

Looks like this bug https://github.com/nats-io/nats-server/issues/2488  is related to your fixes.

**Comment by derekcollison** [comment]
Unclear if they are related tbh.

**Comment by tigrato** [comment]
> Unclear if they are related tbh.

Not sure if it helps but my JetStream was having the same issues described in #2488 and once I switched from file storage to memory storage it improved a lot and no issues until now.

File: server/filestore.go [hunk 1]
```diff
@@ -3126,17 +3082,25 @@ checkCache:
...
 	// We have a cache miss here.
-	if err := mb.loadMsgs(); err != nil {
+	if err := mb.loadMsgsWithLock(); err != nil {
 		return nil, err
 	}
 
-	return mb.cacheLookup(seq)
+	sm, err = mb.cacheLookup(seq)
+	if err == errNoCache {
+		fmt.Printf("NO CACHE [%d] FOR %d!!\n\n", mb.index, seq)
```
Comment (kozlovic) [inline server/filestore.go:3100]: Remove this print, or if needed for debug, make it a log/debug statement.

**Review by kozlovic — CHANGES_REQUESTED** [review]
LGTM, but I believe that there are left over debug print statements.

File: server/filestore.go (hunk 1 above)
Comment (derekcollison) [reply server/filestore.go]: Thanks fixed.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**